src/modules/job-exec/job-exec.c

```diff
@@ -836,6 +841,11 @@ static flux_future_t *jobinfo_start_init (struct jobinfo *job)
     if (!(f_kvs = flux_jobid_kvs_lookup (h, job->id, 0, "jobspec"))
         || flux_future_push (f, "jobspec", f_kvs) < 0)
         goto err;
+    if (job->multiuser
+        && (!(f_kvs = flux_jobid_kvs_lookup (h, job->id, 0, "J"))
+        || flux_future_push (f, "J", f_kvs) < 0)) {
+        goto err;
+    }
```

Comment:
can this code leak `f_kvs` on error?

Reply:
yes, and the other calls in this function seem to suffer the same error. Nice catch!